layer/layer_store.go

```diff
@@ -899,6 +918,74 @@ func (ls *layerStore) deleteUnreferencedDriverLayers(ids []string) int {
 	return total
 }
 
+func (ls *layerStore) findUnusedOverlayFiles() ([]string, error) {
+	var fileNameOverlayIdRegex = regexp.MustCompile("^([a-z0-9]+).*")
+	var overlaysDir = "/var/lib/docker/overlay2"
+
+	files, err := ioutil.ReadDir(overlaysDir)
+	if err != nil {
+		return nil, err
+	}
+
+	fileIDs := make([]string, 0, len(files))
+	// var fileIDs []string
+	for _, info := range files {
+		if match := fileNameOverlayIdRegex.FindStringSubmatch(info.Name()); match != nil {
```

Comment:
we could use `^([a-z0-9]{2,})` to exclude the `l` dir using the regex and then simplify and sanity check like this:

```suggestion
		if info.IsDir() && fileNameOverlayIdRegex.MatchString(info.Name()) {
```

Reply:
:heavy_check_mark: